File: server/api/src/main/kotlin/io/patterueldev/singalong/realtime/RoomCommandType.kt
package io.patterueldev.singalong.realtime

enum class RoomCommandType(val value: String) {
    SKIP_SONG("skipSong"),
    TOGGLE_PLAY_PAUSE("togglePlayPause"),
    ADJUST_VOLUME("adjustVolume"),
    DURATION_UPDATE("durationUpdate"),
    SEEK_DURATION("seekDuration"),
    CANCEL_RESERVATION("cancelReservation"),
    MOVE_RESERVED_SONG_ORDER("moveReservedSongOrder"),
    ;

    companion object {
        fun fromString(value: String): RoomCommandType {
            return entries.first { it.value == value }
        }
    }
}

File: client/packages/singalong_api_client/lib/socket_events/room_command_type.dart
part of '../singalong_api_client.dart';

enum RoomCommandType {
  skipSong,
  togglePlayPause,
  adjustVolume,
  durationUpdate,
  seekDuration,
  cancelReservation,
  moveReservedSongOrder,
  ;

  String get value {
    switch (this) {
      case skipSong:
        return 'skipSong';
      case togglePlayPause:
        return 'togglePlayPause';
      case adjustVolume:
        return 'adjustVolume';
      case durationUpdate:
        return 'durationUpdate';
      case seekDuration:
        return 'seekDuration';
      case cancelReservation:
        return 'cancelReservation';
      case moveReservedSongOrder:
        return 'moveReservedSongOrder';
    }
  }
}
